.github/workflows/check-components-changelog.yml

```diff
@@ -0,0 +1,25 @@
+name: Verify @wordpress/components CHANGELOG update
+
+on:
+    pull_request:
+        types: [opened, synchronize]
+        paths:
+            - 'packages/components/**'
+            - '!packages/components/**/stories/**'
+            - '!packages/components/src**/test/**'
+jobs:
+    check:
+        name: Check CHANGELOG diff
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3 # v3.5.0
+            - name: Fetch origin/trunk
+              run: git fetch --quiet origin trunk
```

Comment:
Not sure if I'm reading this correctly, but it looks like the search for changes to the CHANGELOG file is done against the latest `trunk` (ie. from the `origin` remote). I wonder what happens if the CHANGELOG file is changed on `trunk`?

In my mental model, we should check for differences against the commit that the current PR is based off (something similar to what described in [this article](https://dev.to/scienta/get-changed-files-in-github-actions-1p36)), and not against latest `origin / trunk`

Reply:
Good point, and thanks for the example, it helped a lot.

I've reworked the PR so it should now check against the PR's base commit. While changing this, I also realized I didn't like that we were fetching all of `origin/trunk` when we really only need to go back as far as the base commit, so I've made the fetch depth dynamic to only pull what the given PR needs.